providers/sf_vector_data_provider.py

```diff
@@ -111,36 +111,19 @@ def description(cls) -> str:
 
     @classmethod
     def createProvider(cls, uri, providerOptions, flags=QgsDataProvider.ReadFlags()):
-        return SFVectorDataProvider(uri, providerOptions, flags)
+        base_provider = SFVectorDataProvider(uri, providerOptions, flags)
+        if base_provider._geo_column_type == "NUMBER":
+            return SFH3VectorDataProvider(uri, providerOptions, flags)
+        elif base_provider._geo_column_type in ["GEOGRAPHY", "GEOMETRY"]:
+            return SFGeoVectorDataProvider(uri, providerOptions, flags)
+        else:
+            return base_provider
```

Comment:
In this case the base provider will have the base class extend and featureCount methods implementation? Is this what is expected? 

I am not sure if there is a case where the geo_column_type does not have a NUMBER, GEOGRAPHY, GEOMETRY type but good call to catch it.